### ISSUE TITLE
[Feature] Add YouTube video duration to the summary

### ISSUE THREAD

**Issue by andris155** [FreshRSS/FreshRSS#6722]
**Is your feature request related to a problem? Please describe.**
I usually only have a set amount of time to watch YouTube videos. In such cases, I always have to open the videos to see how long they are and which ones still fit in the time. It would be nice if FreshRSS would show how long the video is in the description.

**Describe the solution you’d like**
The FreshRSS would show how long the video is in the description.

**Describe alternatives you’ve considered**
An extension for this.

**Additional context**
<img width="740" alt="Screenshot 2024-08-19 124356" src="https://github.com/user-attachments/assets/519f6c64-6a25-439b-b367-291c19ee8544">

**Comment by math-GH** [comment]
It is a case for an extension.

Have you checked out these 2 extensions?

- https://github.com/cn-tools/cntools_FreshRssExtensions/tree/master/xExtension-YouTubeChannel2RssFeed (by @cn-tools )
- https://github.com/FreshRSS/Extensions/tree/master/xExtension-YouTube

**Comment by andris155** [comment]
Yes, YouTubeChannel2RssFeed uses the YouTube API to hide short videos, but none of the extensions add option to display video duration time somehow in the summary.

**Comment by cevvalkoala** [comment]
I did something on those lines, and I'll update the extension after some testing.

**Comment by cevvalkoala** [comment]
I added (vibe-added, to be honest) "beginning of summary" as a third duration placement option in YouTube Duration extension. I tested it for a week and it seems to work without problems. Check it out https://github.com/cevvalkoala/xExtension-YouTubeDuration

**Comment by sjlopezb** [comment]
I suggest that the time (in minutes) be displayed against a different coloured background, to avoid confusion with the other text in the video titles. That would be better.

I’ve already enabled it and it works well.

This feature has been available in Inoreader for a year now. It works very well for them, although they don’t have as many options as this new extension; it would be better if it could change the colour of the text and its background so there’s no confusion with anything other than what’s actually visible.

I think it’s a brilliant idea. That way, if I want something bigger or shorter, I can check it out at any time.

Seriously. It works really well.

Cheers!

**Comment by Alkarex** [comment]
Help welcome to contribute to https://github.com/FreshRSS/Extensions/tree/master/xExtension-YouTube

**Comment by sjlopezb** [comment]
> Help welcome to contribute to https://github.com/FreshRSS/Extensions/tree/master/xExtension-YouTube

Hi:

<img width="1433" height="458" alt="Image" src="https://github.com/user-attachments/assets/1a84a677-1ee9-4ca4-8765-319d7e94eb35" />

No problem, I’ve already installed them. I’ve disabled the first one (YoutubeDuration); I’ve disabled the second one because I have lots of YouTube channels that I might need to check to see if there’s anything there I should have for certain situations; and I’ve enabled the third one because there are sites where they embed one or more videos in their posts.

So, it works for me. As for YouTubeDuration, I’ve disabled it because I’ve realised it takes a while to retrieve all the information from the YouTube feeds, for one simple reason: it has to gather the duration of each video from one feed to another, whether from YouTube itself or from the Google API, which I’ve set up via my account to ensure it works for all the videos available. It takes a little over half an hour.

One day I’ll set it up – as soon as I see that there’s another way, apart from the API, to work out the duration of these videos.

In the account where I have all my YouTube feeds, I have them collected every half hour via systemd. So far, not a single YouTube channel has lost any of its content.

Cheers!

**Comment by cevvalkoala** [comment]
> I suggest that the time (in minutes) be displayed against a different coloured background, to avoid confusion with the other text in the video titles. That would be better.

It's a good idea. But first of all, I don't know if it can be done in FreshRSS. I haven't seen any multi-color titles in my feeds.
Secondly, I don't know how to implement it in a safe manner for different themes.

I'll try it sometime, but no promises.
